### PR TITLE
Remove the integrated blocking networking stack

### LINKED ISSUE #2299: move `WifiStack` to seperate crate or use a different stack
Feels a bit odd that it's included in esp-wifi itself, given that our/other BLE stack is separate.

I think we should either extract it, or consider another blocking stack to use for those examples.

### PR DESCRIPTION
**Pull request by bjoernQ**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [x] I have added necessary changes to user code to the [Migration Guide](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/MIGRATING-0.21.md).
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/documentation/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/documentation/CONTRIBUTING.md) and followed its instructions.

#### Description
Closes #2299 

This just removes the blocking networking stack from esp-wifi since it just doesn't belong here. Code is moved to my personal GitHub space for convenience (and I'm fine with maintaining it there)

While it was always possible to use e.g. `smoltcp-nal` or only plain `smoltcp` instead, it wasn't obvious. So, I added one example using `smoltcp-nal` - all others are using the extracted code.

I'm not sure about the `smoltcp-nal` example however - I already see a problem once we see the next `smoltcp` version published and `esp-wifi` (and `embassy-net`) are updated but `smoltcp-nal` isn't yet (which happened a lot in the past).

The options would be
- change that example to use plain `smoltcp`
- just remove that new example

(we could choose one of these options already now or when we run into that situation)

#### Testing
Examples build and work

### CODE REVIEW

**Comment by MabezDev** [comment]
Haven't review this yet, but maybe we could ask (and or PR) smoltcp to split out the driver trait into its own crate `smoltcp-driver` to avoid the upgrade issues. embassy-net doesn't have this issue because the driver trait is a separate crate from the stack. Provided the driver stays the same, it matters much less if smoltcp-nal is out of date, because users can mix and match any version of smoltcp which uses that driver version.

**Comment by bjoernQ** [comment]
> Haven't review this yet, but maybe we could ask (and or PR) smoltcp to split out the driver trait into its own crate `smoltcp-driver` to avoid the upgrade issues. embassy-net doesn't have this issue because the driver trait is a separate crate from the stack. Provided the driver stays the same, it matters much less if smoltcp-nal is out of date, because users can mix and match any version of smoltcp which uses that driver version.

That would be very nice, yes

**Comment by MabezDev** [comment]
I filed https://github.com/smoltcp-rs/smoltcp/issues/1010

**Review by MabezDev — APPROVED** [review]
LGTM, I think it'll be a while before the smoltcp changes land, let's get this out of esp-wifi now.

File: esp-wifi/MIGRATING-0.10.md [hunk 1]
```diff
@@ -42,3 +42,42 @@ The cost of this is that we need to rename the various `async` methods on `WifiC
...
+};
+
+-    let init = init(
+-        timg0.timer0,
+-        Rng::new(peripherals.RNG),
+-        peripherals.RADIO_CLK,
+-    )
+-    .unwrap();
++    let mut rng = Rng::new(peripherals.RNG);
++    let init = init(timg0.timer0, rng.clone(), peripherals.RADIO_CLK).unwrap();
```
Comment (Dominaezzz) [inline esp-wifi/MIGRATING-0.10.md:68]: The diff is a little hard to read. This should make the "What's changed" clearer
```suggestion
+    let mut rng = Rng::new(peripherals.RNG);
-    let init = init(timg0.timer0, rng, peripherals.RADIO_CLK).unwrap();
+    let init = init(timg0.timer0, rng.clone(), peripherals.RADIO_CLK).unwrap();
```

File: esp-wifi/MIGRATING-0.10.md [hunk 2]
```diff
@@ -42,3 +42,42 @@ The cost of this is that we need to rename the various `async` methods on `WifiC
...
+ 
+     let mut wifi = peripherals.WIFI;
++    let (iface, device, mut controller) =
++        create_network_interface(&init, &mut wifi, WifiStaDevice).unwrap();
++
+-    let (iface, device, mut controller, sockets) =
+-        create_network_interface(&init, &mut wifi, WifiStaDevice, &mut socket_set_entries).unwrap();
++    let mut socket_set_entries: [SocketStorage; 3] = Default::default();
```
Comment (Dominaezzz) [inline esp-wifi/MIGRATING-0.10.md:76]: the line is present before *and* after right? In either version this array was still required. Which means you don't need the `+` here, and it should be moved above `create_network_interface`

**Pull request merged**